Automattic/jetpack
# Blocks: load each block from a separate file, take 2

#### Changes proposed in this Pull Request:

As we work to add several new modules, the existing module/blocks.php is updated quite often. It causes a few issues:

- The file keeps getting longer and longer.
- The longer the file, the longer it takes to skim through it.
- Everyone working on that file ends up having to rebase frequently because we are merging changes to that file often.
- Keeping the file in sync with WordPress.com can be more work if someone does not commit their matching WordPress.com diff immediately after merging a Jetpack PR.

Splitting things into different files makes things easier to maintain for everyone.

This commit switches to only loading a block file if:
- Gutenberg is available on the site.
- Gutenpack is enabled on the site.
- the block is available in the block list (index.json)

#### About the chosen structure

I highlighted the new directory structure in a readme file here:
https://github.com/Automattic/jetpack/blob/742e263436dd2b29fd9d7592dd5c2452943f3273/extensions/readme.md

It was chosen so in the future, it can be used to host the blocks' source code as well. When that happens, things should look like this:

```
/extensions/blocks/
/extensions/blocks/map/
/extensions/blocks/map/*.php
/extensions/blocks/map/client/*.jsx
```

If the extension happens to not be a block, but a plugin for example, it can be placed in `/extensions/plugins`.

For further discussion about this, see:
#11286
pafL3P-io-p2

#### Testing instructions:

* Go to Posts > Add New
* Do you see, and can you use all blocks?
* Beta blocks too?
* Yay!

#### Proposed changelog entry for your changes:

* None
